File: kartwheel/app/src/main/java/us/handstand/kartwheel/model/GsonAdapterFactory.java
package us.handstand.kartwheel.model;


import com.google.gson.TypeAdapterFactory;
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory;

@GsonTypeAdapterFactory
public abstract class GsonAdapterFactory implements TypeAdapterFactory {

    // Static factory method to access the package
    // private generated implementation
    public static TypeAdapterFactory create() {
        return new AutoValueGson_GsonAdapterFactory();
    }

}

File: kartwheel/app/src/main/kotlin/us/handstand/kartwheel/model/GsonAdapterFactory.kt
package us.handstand.kartwheel.model


import com.google.gson.TypeAdapterFactory
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory

@GsonTypeAdapterFactory
abstract class GsonAdapterFactory : TypeAdapterFactory {
    companion object {
        // Static factory method to access the package
        // private generated implementation
        @JvmStatic
        fun create(): TypeAdapterFactory {
            return AutoValueGson_GsonAdapterFactory()
        }
    }

}